### DISCUSSION TITLE
Shared library support

### DISCUSSION THREAD

**Discussion by dlinz** [icloud-photos-downloader/icloud_photos_downloader#604] [Ideas]
I tried adding shared library support by replacing /private with /shared in the URL in photos.py, but I got this result:

`
2023-03-13 12:09:45 INFO     Great, you're all set up. The script can now be run without user interaction until 2SA expires.
You can set up email notifications for when the two-step authentication expires.
(Use --help to view information about SMTP options.)
zoneID needs to have ownerRecordName field for calls to sharedb (BAD_REQUEST)
`

I've found my ownerRecordName value as a response to the record query when I initiate it from the browser:

`
{"query":{"recordType":"CPLAssetAndMasterInSmartAlbumByAssetDate","filterBy":[{"fieldName":"startRank","comparator":"EQUALS","fieldValue":{"value":222,"type":"INT64"}},{"fieldName":"direction","comparator":"EQUALS","fieldValue":{"value":"DESCENDING","type":"STRING"}},{"fieldName":"smartAlbum","comparator":"EQUALS","fieldValue":{"type":"STRING","value":"VIDEO"}}]},"zoneID":{"zoneName":"SharedSync-A4043105-9005-46CD-9D76-C6D045EAB757","ownerRecordName":"_00f8c71d43dda5bcd0ac689fd15e64dc","zoneType":"REGULAR_CUSTOM_ZONE"},"desiredKeys":["addedDate","adjustmentRenderType","adjustmentType","assetDate","assetHDRType","assetSubtype","assetSubtypeV2","burstFlags","burstFlagsExt","burstId","captionEnc","codec","contributors","created","customRenderedValue","dataClassType","dateExpunged","duration","extendedDescEnc","filenameEnc","importedBy","importedByBundleIdentifierEnc","importedByDisplayNameEnc","isDeleted","isExpunged","isFavorite","isHidden","isSparsePrivateRecord","itemType","linkedShareRecordName","linkedShareZoneName","linkedShareZoneOwner","locationEnc","locationLatitude","locationLongitude","locationV2Enc","masterRef","mediaMetaDataEnc","mediaMetaDataType","orientation","originalOrientation","recordChangeTag","recordName","recordType","remappedBy","remappedRef","resJPEGFullFileType","resJPEGFullFingerprint","resJPEGFullHeight","resJPEGFullRes","resJPEGFullWidth","resJPEGLargeFileType","resJPEGLargeFingerprint","resJPEGLargeHeight","resJPEGLargeRes","resJPEGLargeWidth","resJPEGMedFileType","resJPEGMedFingerprint","resJPEGMedHeight","resJPEGMedRes","resJPEGMedWidth","resJPEGThumbFileType","resJPEGThumbFingerprint","resJPEGThumbHeight","resJPEGThumbRes","resJPEGThumbWidth","resOriginalAltFileType","resOriginalAltFingerprint","resOriginalAltHeight","resOriginalAltRes","resOriginalAltWidth","resOriginalFileType","resOriginalFingerprint","resOriginalHeight","resOriginalRes","resOriginalVidComplFileType","resOriginalVidComplFingerprint","resOriginalVidComplHeight","resOriginalVidComplRes","resOriginalVidComplWidth","resOriginalWidth","resSidecarFileType","resSidecarFingerprint","resSidecarHeight","resSidecarRes","resSidecarWidth","resVidFullFileType","resVidFullFingerprint","resVidFullHeight","resVidFullRes","resVidFullWidth","resVidHDRMedRes","resVidMedFileType","resVidMedFingerprint","resVidMedHeight","resVidMedRes","resVidMedWidth","resVidSmallFileType","resVidSmallFingerprint","resVidSmallHeight","resVidSmallRes","resVidSmallWidth","timeZoneOffset","vidComplDispScale","vidComplDispValue","vidComplDurScale","vidComplDurValue","vidComplVisibilityState","videoFrameRate","zoneID"],"resultsLimit":60}`

I'm guessing I need to edit the query syntax to change the zoneID from primarySync to SharedSync and add ownerRecordName to the parameters, but maybe someone who knows python could help?

Thanks!

**Comment by andhou** [comment]
I really need this - hope you figure out how to implement it.
It seems to be possible via docker-icloudpd (https://github.com/boredazfcuk/docker-icloudpd)

**Comment by coffeemug13** [comment]
Please, what's the status about this feature? It looks like, that there was a solution in pyicloud, but this library is not used in pyicloud-pd.  I guess, downloading the shared library is for most Apple User a high priority.

**Reply by AndreyNikiforov** [reply]
shared libs are supported since 1.16.0

**Reply by coffeemug13** [reply]
Thanks, found it. Just an idea: add an example in the Readme?

**Reply by AndreyNikiforov** [reply]
Great idea! Feel free to add a note about shared libs to the readme
